src/sentry_transport.c

```diff
@@ -165,9 +167,24 @@ sentry__prepare_http_request(sentry_envelope_t *envelope,
         return NULL;
     }
 
+    char *compressed_body = NULL;
+    size_t compressed_body_len = 0;
+    bool compressed = sentry_gzipped_with_compression(
+        body, body_len, &compressed_body, &compressed_body_len);
+    if (compressed) {
+        if (body_owned) {
+            sentry_free(body);
+            body = NULL;
+            body_owned = false;
+        }
+    }
```

Comment:
Once you compress the body, you do not need the original `body` anymore.
Therefore, I would just reuse the `body/_owned` variables so you don’t have to duplicate all the code down below.

Reply:
Yes, if enabled compress,  original body need free.